# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/colo/hotfix.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- colo-1.22/stage2/src/heap.c.vanilla	2025-05-16 16:59:04.679811132 +0200
+++ colo-1.22/stage2/src/heap.c	2025-05-16 16:59:42.064906916 +0200
@@ -25,6 +25,7 @@
 void heap_reset(void)
 {
 	extern char __text;
+#define restrict res
 	void *restrict;
 
 	assert(!((unsigned long) &__text & 15));
@@ -35,7 +36,7 @@
 	restrict = KSEG0(ram_restrict) - (16 << 10);	// XXX
 	if(free_hi > restrict)
 		free_hi = restrict;
-
+#undef restrict
 	image_size = 0;
 	image_size_mark = 0;
 
--- colo-1.22/stage2/src/shell.c.vanilla	2025-05-16 16:58:20.025608869 +0200
+++ colo-1.22/stage2/src/shell.c	2025-05-16 16:58:44.771193447 +0200
@@ -243,6 +243,8 @@
 		*(volatile uint8_t *) BRDG_NCS0_BASE = leds;
 		udelay(400000);
 	}
+
+	return E_NONE;
 }
 
 /*
--- colo-1.22/stage2/src/lib.c.vanilla	2025-05-16 16:57:34.741856582 +0200
+++ colo-1.22/stage2/src/lib.c	2025-05-16 16:58:03.268307836 +0200
@@ -60,9 +60,7 @@
 
 char *strcpy(char *dst, const char *src)
 {
-	char *ptr;
-
-	for(ptr = dst; (*dst++ = *src++);)
+	for(; (*dst++ = *src++);)
 		;
 
 	return dst;
